homeassistant/components/sms/__init__.py

```diff
@@ -44,13 +47,27 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     if not gateway:
         return False
     hass.data[DOMAIN][SMS_GATEWAY] = gateway
```

Comment:
Don't use a constant but instead use `entry.entry_id` to store the gateway. That way you can support multiple config flows.

Reply:
> Don't use a constant but instead use `entry.entry_id` to store the gateway. That way you can support multiple config flows.

We cannot use the entry id as we need to obtain the gateway from the notify service, but notify services does not support config flow yet.

Reply:
Ah, too bad 😢